packages/apollo-server-core/src/types.ts

```diff
@@ -54,12 +55,28 @@ export type GraphQLServiceConfig = {
 
 export interface GatewayInterface {
   load(options: { apollo: ApolloConfig }): Promise<GraphQLServiceConfig>;
+
+  /**
+   * @deprecated Use `onSchemaLoadOrUpdate` instead
```

Comment:
Let's make this optional so we are eventually able to remove it?

Reply:
I'd be down for this. It would cause TS failures for people compiling against `GatewayInterface`, but this should be few (if any) people, and it's arguably a decent way of notifying them that this method is disappearing soon (and that they should change their code accordingly).